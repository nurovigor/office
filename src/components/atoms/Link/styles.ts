import { css } from 'linaria';
import { colors } from 'src/components/_settings/_variables';

export const LinkStyle = css`
	text-decoration: none;
	color: ${colors.white};
	font-size: 20px;
	background: #1d3557;
	border: none;
	cursor: pointer;
	padding: 5px 10px;
	transition: 0.3s;
	&.Active {
		background: ${colors.success};
	}
	&:hover {
		background-color: ${colors.success};
	}
`;
