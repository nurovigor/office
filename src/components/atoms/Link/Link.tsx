import React from 'react';
import { Link as NavLink } from 'react-router-dom';
import { LinkStyle } from './styles';
import cn from 'classnames';
type LinkPropsType = {
	path: string;
	title: string;
	isActive: boolean;
};

export const Link: React.FC<LinkPropsType> = ({ path, title, isActive }) => {
	return (
		<NavLink
			className={cn(LinkStyle, {
				Active: isActive
			})}
			to={path}
		>
			{title}
		</NavLink>
	);
};
