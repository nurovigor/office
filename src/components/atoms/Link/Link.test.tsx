import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Link } from './Link';

describe('renders Link component', () => {
	it('Link render with title', () => {
		render(
			<Router>
				<Link path="#" isActive={false} title="Link" />
			</Router>
		);
		expect(screen.getByText(/Link/i)).toBeInTheDocument();
	});

	it('Link snapshot', () => {
		const { asFragment } = render(
			<Router>
				<Link path="#" isActive={false} title="Link" />
			</Router>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
