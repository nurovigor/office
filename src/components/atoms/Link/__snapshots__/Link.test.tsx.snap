// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Link component Link snapshot 1`] = `
<DocumentFragment>
  <a
    class=""
    href="/"
  >
    Link
  </a>
</DocumentFragment>
`;
